typing/mode_intf.mli

```diff
@@ -373,16 +373,18 @@ module type S = sig
 
   module Axis : sig
     (** ('p, 'r) t represents a projection from a product of type ['p] to an
-    element of type ['r]. *)
+    element of type ['r].
+
+    NB: must listed in the order of axis implication. See [typemode.ml]. *)
```

Comment:
Why?

Reply:
because `transl_modalities` relies on this fact to have implied modalities correct. If user write `yielding global`, we first reorder it to `global yielding`, so we see `yielding` later than `global` and override the `unyielding` implied by `global`.

Reply:
Oh, right, I see.